File: src/main/java/com/sfh/pokeRogueBot/phase/impl/EvolutionPhase.java
package com.sfh.pokeRogueBot.phase.impl;

import com.sfh.pokeRogueBot.model.enums.UiMode;
import com.sfh.pokeRogueBot.model.exception.NotSupportedException;
import com.sfh.pokeRogueBot.phase.AbstractPhase;
import com.sfh.pokeRogueBot.phase.Phase;
import com.sfh.pokeRogueBot.phase.actions.PhaseAction;
import org.springframework.stereotype.Component;

@Component
public class EvolutionPhase extends AbstractPhase implements Phase {

    public static final String NAME = "EvolutionPhase";

    @Override
    public String getPhaseName() {
        return NAME;
    }

    @Override
    public PhaseAction[] getActionsForUiMode(UiMode uiMode) throws NotSupportedException {
        if (uiMode == UiMode.MESSAGE) {
            return new PhaseAction[]{
                    this.pressSpace
            };
        } else if (uiMode == UiMode.EVOLUTION_SCENE) {
            return new PhaseAction[]{
                    this.waitBriefly,
                    this.pressSpace
            };
        }

        throw new NotSupportedException("GameMode " + uiMode + " is not supported in " + NAME);
    }
}

File: src/main/kotlin/com/sfh/pokeRogueBot/phase/impl/EvolutionPhase.kt
package com.sfh.pokeRogueBot.phase.impl

import com.sfh.pokeRogueBot.model.enums.UiMode
import com.sfh.pokeRogueBot.model.exception.UnsupportedUiModeException
import com.sfh.pokeRogueBot.phase.UiPhase
import com.sfh.pokeRogueBot.service.javascript.JsUiService
import org.springframework.stereotype.Component

@Component
class EvolutionPhase(
    private val jsUiService: JsUiService,
) : UiPhase {
    override val phaseName = "EvolutionPhase"

    override fun handleUiMode(uiMode: UiMode) {
        when (uiMode) {
            UiMode.EVOLUTION_SCENE -> {
                jsUiService.sendActionButton()
            }

            else -> throw UnsupportedUiModeException(uiMode)
        }
    }
}